frontend/src/metabase-lib/v1/expressions/recursive-parser.js

```diff
@@ -394,6 +394,15 @@ export const adjustBooleans = tree =>
     return node;
   });
 
+export const adjustTopLevelLiteral = tree => {
+  const type = typeof tree;
+  if (type === "string" || type === "number" || type === "boolean") {
```

Comment:
Could we use the helpers from `./matchers` here?

Reply:
Fixed